overlord/snapstate/snapstate.go

```diff
@@ -820,6 +825,7 @@ func doUpdate(ctx context.Context, st *state.State, names []string, updates []*s
 		}
 	}
 
+	var rerefreshes []string
```

Comment:
I think precomputing this set here feels a bit too constraining in terms of the logic we want after a refresh going forward, I think we should always have a re-refresh task now, but then it you should only (for now) consider things that went through an epoch transition and would again